src/lib/MathExt.sol

```diff
@@ -1,17 +1,52 @@
 pragma solidity ^0.8.0;
 
+import {SafeCast} from "openzeppelin-contracts/utils/math/SafeCast.sol";
+import {Math} from "openzeppelin-contracts/utils/math/Math.sol";
+
 error NegativeUint();
 
-function sub(uint256 a, int256 b) pure returns (uint256) {
-    if (b < 0) {
-        // If b is negative, add its absolute value to a
-        return a + uint256(-b);
-    } else {
-        // If b is positive, subtract it from a
-        if (a < uint256(b)) {
-            revert NegativeUint();
+library MathExt {
+    function sub(uint256 a, int256 b) internal pure returns (uint256) {
+        if (b < 0) {
+            // If b is negative, add its absolute value to a
+            return a + uint256(-b);
+        } else {
+            // If b is positive, subtract it from a
+            if (a < uint256(b)) {
+                revert NegativeUint();
+            }
+
+            return a - uint256(b);
+        }
+    }
+
+    function boundedSub(uint256 a, int256 b, uint256 min, uint256 max) internal pure returns (uint256 r) {
```

Comment:
i think it would be nice to introduce another function in this library "bound", which applies the bounding logic at the end of this function